src/transformStateWithClones.js

```diff
@@ -7,7 +7,53 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrayObjects = [];
+  let currentState = {};
+
+  for (const object of actions) {
+    if (arrayObjects.length === 0) {
+      currentState = state;
+    } else {
+      currentState = arrayObjects[arrayObjects.length - 1];
+    }
+
+    if (object.type === 'addProperties') {
+      currentState = {
+        ...currentState, ...object.extraData,
+      };
+      arrayObjects.push(currentState);
```

Comment:
```suggestion
      arrayObjects.push({ ...currentState });
```
При добавлении объекта в массив можно делать его копию, чтобы получался clone